tensorflow-nn/src/TensorFlow/NN.hs

```diff
@@ -73,13 +73,12 @@ sigmoidCrossEntropyWithLogits
   :: (OneOf '[Float, Double] a, TensorType a, Num a)
      => Tensor Value a          -- ^ __logits__
      -> Tensor Value a          -- ^ __targets__
-     -> Build (Tensor Value a)
+     -> Tensor Value a
 sigmoidCrossEntropyWithLogits logits targets = do
     let zeros = zerosLike logits
         cond = logits `greaterEqual` zeros
         relu_logits = select cond logits zeros
         neg_abs_logits = select cond (-logits) logits
-    withNameScope "logistic_loss" $ do
```

Comment:
Here is a version that works with `Build`:

```haskell
sigmoidCrossEntropyWithLogits
  :: (OneOf '[Float, Double] a, TensorType a, Num a)
     => Tensor Value a          -- ^ __logits__
     -> Tensor Value a          -- ^ __targets__
     -> Build (Tensor Value a)
sigmoidCrossEntropyWithLogits logits targets = do
    logits' <- render logits
    targets' <- render targets
    let zeros = zerosLike logits'
        cond = logits' `greaterEqual` zeros
        relu_logits = select cond logits' zeros
        neg_abs_logits = select cond (-logits') logits'
    withNameScope "logistic_loss" $ do
        left  <- render $ relu_logits - logits' * targets'
        right <- render $ log (1 + exp neg_abs_logits)
        withNameScope "sigmoid_add" $ render $ left `add` right
```

(the only difference is that it renders the input tensors outside of the scope)

This is one of the confusing aspects of how we are handling name scopes right now. The `logits` here is rendered inside the "logistic_loss" name scope and the one in the test (`vLogits` in `TF.gradients tfLoss [vLogits]`) is rendered outside of it, so you end up with two distinct tensors. The one in the test is never used, so the gradient w.r.t. it is always zero.

@blackgnezdo probably didn't run into this because we usually only calculate for the gradient w.r.t. a variable and variable creation is done inside `Build`.